src/snowflake/cli/plugins/nativeapp/artifacts.py

```diff
@@ -283,3 +283,27 @@ def find_version_info_in_manifest_file(
             patch_name = version_info[patch_field]
 
     return version_name, patch_name
+
+
+def _determine_artifacts_file_path(
+    source: PurePath, artifacts: List[ArtifactMapping]
+) -> Optional[str]:
+    """Given a source file path (relative to project root) that doesn't necessarily exist on the file system, returns the destination path string for the first matching artifact (relative to deploy root), or None if none matched."""
+    for artifact in artifacts:
+        source_path = PurePath(source)
```

Comment:
Nit: you should be able to use `source` directly as is, why do we need source_path?